mlflow/sklearn/__init__.py

```diff
@@ -1190,6 +1225,31 @@ def fetch_logged_data(run_id):
             stacklevel=2,
         )
 
+    def fit_mlflow_xgboost(original, self, *args, **kwargs):
+        """
+        Autologging function for XGBoost scikit-learn models
+        """
+        # parameter, metric, and non-model artifact logging
+        # are done in `train()` in `mlflow.xgboost.autolog()`
+        fit_output = original(self, *args, **kwargs)
+        # log models after training
+        (X, _, _) = _get_args_for_metrics(self.fit, args, kwargs)
```

Comment:
Does `_get_args_for_metrics` always return a tuple with 3 elements?

Reply:
Never mind. It does:

https://github.com/mlflow/mlflow/blob/22e69c3bdba7c4af616f6f0ce4e74821dc665b10/mlflow/sklearn/utils.py#L48

Reply:
`X = _get_args_for_metrics(self.fit, args, kwargs)[0]` might be safer.

Reply:
We might want to rename `_get_args_for_metrics` to something like `_get_X_y_and_sample_weight`. I'll take care of this.